packages/core/tests/package/dependencies/TransitiveDependencyResolver.test.ts

```diff
@@ -0,0 +1,132 @@
+import { jest, expect } from '@jest/globals';
+import { MockTestOrgData, testSetup } from '@salesforce/core/lib/testSetup';
+import { Connection, AuthInfo } from '@salesforce/core';
+import TransitiveDependencyResolver from '../../../src/dependency/TransitiveDependencyResolver';
+const $$ = testSetup();
+
+const setupFakeConnection = async () => {
+  const testData = new MockTestOrgData();
+  testData.makeDevHub();
+
+  $$.setConfigStubContents('AuthInfoConfig', {
+      contents: await testData.getConfig(),
+  });
+  $$.fakeConnectionRequest = (request) => {
+    return Promise.resolve(response);
+  };
+
+  const conn = await Connection.create({
+    authInfo: await AuthInfo.create({username: testData.username})
+  });
+
+  return conn;
+}
+
+jest.mock('../../../../core/src/git/Git', () => {
+  class Git {
+    static async initiateRepo()
+     {
+      return new Git();
+     }
+  }
+
+  return Git;
+});
+
+jest.mock('../../../../core/src/git/GitTags', () => {
+  class GitTags {
+      async listTagsOnBranch(): Promise<string[]> {
+          return gitTags;
+      }
+  }
+
+  return GitTags;
+});
+
+let conn: Connection;
+let gitTags;
+let response;
+
+describe("Given a TransitiveDependencyResolver", () => {
+
+  beforeEach(async () => {
+    conn = await setupFakeConnection();
+
+  })
+
+  it("should resolve missing package dependencies with transitive dependency", async () => {
+    const transitiveDependencyResolver = new TransitiveDependencyResolver(projectConfig, conn);
+    const resolvedProjectConfig: any = await transitiveDependencyResolver.exec();
+
+    let packageDescriptor = resolvedProjectConfig.packageDirectories.find((dir) => dir.package === "candidate-management");
+    let coreDependency = packageDescriptor.dependencies.find(dependency => dependency.package === "temp");
```

Comment:
I didn't understand what the test does here,

Reply:
This is to check whether the missing dependencies are added, and the version is same as its dependent pacakge